mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go.erb

```diff
@@ -1597,10 +1674,46 @@ func resourceBigQueryTableRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	<% unless version == 'ga' -%>
+	// Get API fields for TableReplicationInfo not available in the client library yet.
+	url, err := tpgresource.ReplaceVars(d, config, "{{BigQueryBasePath}}projects/{{project}}/datasets/{{dataset_id}}/tables/{{table_id}}")
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Reading BigQuery table through API: %s", url)
+
+	getRes, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config:    config,
+		Method:    "GET",
+		RawURL:    url,
+		UserAgent: userAgent,
+	})
+	if err != nil {
+		return err
+	}
+
+	if v, ok := getRes["tableReplicationInfo"]; ok {
+		tableReplicationInfo := flattenTableReplicationInfo(v.(map[string]interface{}))
+
+		if err := d.Set("table_replication_info", tableReplicationInfo); err != nil {
+			return fmt.Errorf("Error setting table replication info: %s", err)
+		}
+	}
+
+  <% end -%>
 	return nil
 }
 
 func resourceBigQueryTableUpdate(d *schema.ResourceData, meta interface{}) error {
+	<% unless version == 'ga' -%>
+	if v, ok := d.GetOk("table_replication_info"); ok {
```

Comment:
Hmm, I'm wondering if this covers enough edge cases to prevent users from getting into a weird state. For example, do you know if it is possible/intended for a user to specify these fields, and then remove them with an update (ie. turn off replication by replacing the resource)?

Alternatively, could these fields be marked as `ForceNew: true`? That would also prevent in-place updates, but still allow Terraform users to make changes, which would trigger the resource will be re-created. I think this would match Terraform usage expectations better, but there might be downsides to that approach for this use case.

Reply:
Actually `ForceNew: true` would make sense, as long as removing it in the future when Update support does become available won't be considered a breaking change (shouldn't be since it's going from immutable to mutable). Added this to the schema and removed the Update unit test.

Reply:
Yep, adding Update support later would not be a breaking change.

I think you may want to add `ForceNew: true` to `table_replication_info` directly as well. I cannot remember all of the edges of the behavior, but I think there a few ways to change `table_replication_info` that would not trigger the ForceNew of the child fields, and I don't believe you want _any_ changes to `table_replication_info` to trigger an update.